search/topbottom/topbottom.md

```diff
@@ -0,0 +1,77 @@
+# Top / Bottom Modules
+
+The `top` and `bottom` modules show the top or bottom N values of a given enumerated value (or values) or entry. For example, to show the top 10 values of the enumerated value `foo`:
+
+```gravwell
+tag=foo json foo | top -n 10 foo
+```
+
+The output is sorted in descending order (ascending for the `bottom` module).
+
+The modules use the same syntax. Each optionally takes one or more enumerated values. The modules will emit the top/bottom values of the given value. If no enumerated value is provided, the `DATA` field is used.
+
+If multiple enumerated values are provided, the modules will sort by the top/bottom values of the enumerated values, in the provided order. If a two entries have the same value for a given enumerated value, the second enumerated value will be used.
+
+For example, given:
+
+| foo | bar |
+|-----|-----|
+| 1   |  1  |
+| 10  | 9   |
+| 10  | 200 |
+| 10  | 100 |
+
+and the query: 
+
+```gravwell
+tag=foo json foo bar | top -n 2 foo bar
+```
+
+The output will be:
+
+| foo | bar |
+|-----|-----|
+| 10  | 200 | 
+| 10  | 100 |
+
+In the example above, `top` took the top two values of `foo` first, and where `foo` had the same value, took the top value of `bar` instead.
```

Comment:
"The output will be:

| foo | bar |
|-----|-----|
| 10  | 200 | 
| 10  | 100 |

In the example above, `top` took the top two values of `foo` first, and where `foo` had the same value, took the top value of `bar` instead."

I was a little but confused by this example. As I understand it, the small foo bar table in this example is sort of an intermediate step that shows how the processing works but the _**output**_ will be 10, 200 - just those two values. I would change the language here to resent this small table as an intermediate step (what's happening under the hood) and then say "so the final result will be: "

Reply:
No -- the returned data will be two entries. And those two entries each have a foo and bar value. And these are the values on the entries. Additionally it will come back in that order.

Reply:
Gotcha! ok. I was still really confused when I read this, so maybe we can clarify in some way?

Reply:
Maybe describe `bar` as a tie breaker. Saying "instead" makes me feel like `foo` should be discarded in favor of `bar`.